language/vm/vm-runtime/src/gas_meter.rs

```diff
@@ -2,19 +2,89 @@
 // SPDX-License-Identifier: Apache-2.0
 
 //! Gas metering logic for the Move VM.
-use crate::{interpreter::InterpreterForGasCost, loaded_data::function::FunctionReference};
+use crate::{
+    code_cache::module_cache::ModuleCache,
+    data_cache::RemoteCache,
+    identifier::{create_access_path, resource_storage_key},
+    interpreter::InterpreterForGasCost,
+    loaded_data::function::FunctionReference,
+};
 use libra_types::{
     account_address::ADDRESS_LENGTH,
+    account_config,
+    identifier::Identifier,
+    language_storage::ModuleId,
     transaction::MAX_TRANSACTION_SIZE_IN_BYTES,
-    vm_error::{StatusCode, VMStatus},
+    vm_error::{sub_status, StatusCode, VMStatus},
+};
+use vm::{
+    access::ModuleAccess,
+    errors::VMResult,
+    file_format::{Bytecode, StructDefinitionIndex},
+    gas_schedule::*,
 };
-use vm::{access::ModuleAccess, errors::VMResult, file_format::Bytecode, gas_schedule::*};
+
+//***************************************************************************
+// Gas Schedule Loading
+//***************************************************************************
+
+lazy_static! {
+    /// The ModuleId for the gas schedule module
+    pub static ref GAS_SCHEDULE_MODULE: ModuleId =
+        { ModuleId::new(account_config::core_code_address(), Identifier::new("GasSchedule").unwrap()) };
+}
+
+pub(crate) fn load_gas_schedule<'alloc, P>(
+    module_cache: P,
+    data_view: &dyn RemoteCache,
+) -> VMResult<CostTable>
+where
+    P: ModuleCache<'alloc>,
+{
+    let address = account_config::association_address();
+    let gas_module = module_cache
+        .get_loaded_module(&GAS_SCHEDULE_MODULE)?
+        .ok_or_else(|| {
+            VMStatus::new(StatusCode::GAS_SCHEDULE_ERROR)
+                .with_sub_status(sub_status::GSE_UNABLE_TO_LOAD_MODULE)
+        })?;
+    // NB: The gas schedule resource is at the 1'st struct handle index in the module -- in
+    // particular since it is defined after the GasCost struct.
+    let struct_tag = resource_storage_key(
+        gas_module,
+        StructDefinitionIndex::new(GAS_SCHEDULE_RESOURCE_DEF_IDX),
```

Comment:
I don't quite follow. We can't resolve by name because we need to get the data from a particular resource published on chain. I might just be doing it wrong though?